crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -417,13 +432,41 @@ impl DecisionTree {
         }
     }
 
+    fn short_circuit(ctx: &mut SsaContext, stack: &mut StackFrame, condition: NodeId) -> bool {
+        if ctx.under_assumption(condition) {
+            block::short_circuit_instructions(ctx, &stack.stack);
+            let nop = stack.stack[0];
+            stack.stack.clear();
+            stack.stack.push(nop);
+            let operation =
+                Operation::Cond { condition, val_true: ctx.zero(), val_false: ctx.one() };
+            let cond = ctx.add_instruction(Instruction::new(
+                operation,
+                ObjectType::Boolean,
+                Some(stack.block),
+            ));
+            stack.push(cond);
+            let unreachable = Operation::Constrain(cond, None);
```

Comment:
Perhaps we could avoid the None error location here if Blocks had locations. That way we could report the block as being unreachable and avoid the Option<Location>.

Reply:
Yes I was thinking about this as well, however I believe this would come in a separate PR.